metis-common/metis-common-zoho/src/main/java/eu/europeana/metis/zoho/ZohoAccessClient.java

```diff
@@ -100,20 +114,50 @@ public ZohoAccessClient(String grantToken, Map<String, String> zcrmConfiguration
    * @return the organization corresponding to the organization name
    * @throws ZohoException if an exception occurred during searching
    */
-  public Optional<ZCRMRecord> getZcrmRecordOrganizationByName(String organizationName)
+  public Optional<Record> getZohoRecordOrganizationByName(String organizationName)
       throws ZohoException {
-    final BulkAPIResponse bulkAPIResponseAccounts;
     try {
-      bulkAPIResponseAccounts = zcrmModuleAccounts.searchByCriteria(String
+      RecordOperations recordOperations = new RecordOperations();
+      ParameterMap paramInstance = new ParameterMap();
+      paramInstance.add(SearchRecordsParam.CRITERIA, String
           .format(ZohoConstants.ZOHO_OPERATION_FORMAT_STRING, ZohoConstants.ACCOUNT_NAME_FIELD,
               ZohoConstants.EQUALS_OPERATION, organizationName));
-    } catch (ZCRMException e) {
+
+      APIResponse<ResponseHandler> response = recordOperations
+          .searchRecords(ZohoConstants.ACCOUNTS_MODULE_NAME, paramInstance);
+      return getZohoRecord(response);
```

Comment:
It looks like multiple methods in this class are almost the same apart from the specific configuration of the paramInstance. Is there some generalization that we can do?